lib/media/adaptation_set_criteria.js

```diff
@@ -267,4 +269,41 @@ shaka.media.PreferenceBasedCriteria = class {
       return true;
     });
   }
+
+  /**
+   * Filter variants based on audio group ID of
+   * the highest bandwidth variant, if present
+   * @param {!Array.<shaka.extern.Variant>} variants
+   * @return {!Array.<shaka.extern.Variant>}
+   * @private
+   */
+  static filterByAudioGroupId_(variants) {
+    const Class = shaka.media.PreferenceBasedCriteria;
+    const bestVariant = Class.getBestVariantByBandwidth_(variants);
+
+    if (bestVariant && bestVariant.audio &&
+        bestVariant.audio.groupId !== undefined) {
+      return variants.filter((variant) => {
+        return variant.audio.groupId === bestVariant.audio.groupId;
```

Comment:
It seems like this is handled already if you fix AdaptationSet.  We call add(), which calls canInclude(), which calls areAdaptable(), which calls areAudiosCompatible_().  And the constructor of AdaptationSet calls add().  So creating an AdaptationSet all at once or in pieces should suffice without extra filtering logic here.

I believe if you revert this file and make the changes I recommended in AdaptationSet, your new test case should still cover the desired outcome and tell you if I'm wrong.  :smile:

Reply:
Yes, this simplifies everything, and all because of a mistake.